api/pkg/apis/v1alpha1/managers/configs/configs-manager.go

```diff
@@ -102,46 +112,54 @@ func (s *ConfigsManager) Get(object string, field string, overlays []string, loc
 	for _, key := range s.Precedence {
 		if provider, ok := s.ConfigProviders[key]; ok {
 			if field == "" {
-				configObj, err := s.getObjectWithOverlay(provider, object, overlays, localContext)
+				configObj, err := s.getObjectWithOverlay(ctx, provider, object, overlays, localContext)
 				if err != nil {
 					return "", err
 				}
 				return configObj, nil
 			} else {
-				if value, err := s.getWithOverlay(provider, object, field, overlays, localContext); err == nil {
+				if value, err := s.getWithOverlay(ctx, provider, object, field, overlays, localContext); err == nil {
 					return value, nil
 				}
 			}
 		}
 	}
-	return "", v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid config object or key: %s, %s", object, field), v1alpha2.BadRequest)
+	err = v1alpha2.NewCOAError(nil, fmt.Sprintf("Invalid config object or key: %s, %s", object, field), v1alpha2.BadRequest)
+	return "", err
 }
-func (s *ConfigsManager) getWithOverlay(provider config.IConfigProvider, object string, field string, overlays []string, localContext interface{}) (interface{}, error) {
+func (s *ConfigsManager) getWithOverlay(ctx context.Context, provider config.IConfigProvider, object string, field string, overlays []string, localContext interface{}) (interface{}, error) {
 	if len(overlays) > 0 {
 		for _, overlay := range overlays {
-			if overlayObject, err := provider.Read(overlay, field, localContext); err == nil {
+			if overlayObject, err := provider.Read(ctx, overlay, field, localContext); err == nil {
 				return overlayObject, nil
 			}
 		}
 	}
-	return provider.Read(object, field, localContext)
+	return provider.Read(ctx, object, field, localContext)
 }
 
 func (s *ConfigsManager) GetObject(object string, overlays []string, localContext interface{}) (map[string]interface{}, error) {
-	log.Debugf(" M (Config): GetObject %v, config provider size %d", object, len(s.ConfigProviders))
+	ctx, span := observability.StartSpan("Config Manager", context.TODO(), &map[string]string{
```

Comment:
The method methods need to take a context parameter, which is passed in by the vendor

Reply:
@iwangjintian , how much effort you think if we can change the method signature to accept the context parameter in config/secret mgrs and providers? If the effort is not that much, please change in this PR. If it is too big, please log an issue then complete this PR at first and propose a new PR to fix this. And this fix should be before public preview.

Reply:
I‘ve opened an issue for this: https://github.com/eclipse-symphony/symphony/issues/430
Like we discussed, the point here is using evaluationContext to provide the context.Context since config manager and secret manager is not only called by vendor but also the parser. Parser is not keeping the context.Context.
@linyguo will be working on this.